File: Qb.Js/src/Join.ts
import {FromTerm} from "./FromTerm";
import {WhereClause} from "./WhereClause";
import {JoinType} from "./Enums";



interface IJoin {
  LeftTable: FromTerm;
  Conditions: WhereClause;
  RightTable: FromTerm;
  Type: JoinType;
}

export class Join implements IJoin {
  LeftTable: FromTerm;
  Conditions: WhereClause;
  RightTable: FromTerm;
  Type: JoinType;

  constructor(leftTable: FromTerm, rightTable: FromTerm, conditions: WhereClause, type: JoinType) {
    this.LeftTable = leftTable;
    this.RightTable = rightTable;
    this.Conditions = conditions;
    this.Type = type;
  }
}

File: Qb.Net/SqlOm/Join.cs
namespace Viten.QueryBuilder.SqlOm
{
  /// <summary>
  /// Represnts a Join between two tables.
  /// </summary>
  public class Join
	{
    /// <summary></summary>
    public FromTerm LeftTable { get; set; }
    /// <summary></summary>
    public FromTerm RightTable { get; set; }

    /// <summary></summary>
    public WhereClause Conditions { get; set; }

    /// <summary></summary>
    public JoinType Type { get; set; }

    public Join()
    {
    }

    /// <summary></summary>
    public Join(FromTerm leftTable, FromTerm rightTable, WhereClause conditions, JoinType type)
		{
			LeftTable = leftTable;
			RightTable = rightTable;
			Conditions = conditions;
			Type = type;
		}


  }
}
